main.py

```diff
@@ -218,6 +219,9 @@ def ui_access_wrapper(self, *args, **kwargs):
     get_redirect_route(r'/forum', pages.ForumRedirectPage),
 
     get_redirect_route(r'%s' % feconf.ADMIN_URL, admin.AdminPage),
+    get_redirect_route(
+        r'/initialize_android_test_data',
+        android_e2e_config.InitializeAndroidTestData),
```

Comment:
@FareesHussain Is this API is expected and ready for use in production?

Reply:
It is not for production, I've added the flag for that
https://github.com/oppia/oppia/blob/5f1577e489ce9eb7c31cf048caa7402ba2603d04/core/controllers/android_e2e_config.py#L50

Reply:
Ideally we would only register this in dev mode runs of the codebase, but per @vojtechjelinek it seemed that there wasn't an easy way to do that.

Reply:
@BenHenning, Do you want this URL to be registered only for DEV_MODE? If yes, then can we do the following [around line 885](https://github.com/oppia/oppia/blob/develop/main.py#L885)
```
if constants.DEV_MODE: 
    URLS.append(
      get_redirect_route(
        r'/initialize_android_test_data',
        android_e2e_config.InitializeAndroidTestData))
```

Reply:
@DubeySandeep Let me know if I'm wrong at some point

Using this in production mode shall give a 404 error right.
I tried this and ran the tests I've written
https://github.com/oppia/oppia/blob/b1889f7044a51f47b3a92e0ce25e88db5ee7f05d/core/controllers/android_e2e_config_test.py#L55

If it is production mode, this URL shouldn't exist and gives a 404 
but the test passes (showing the exception `Cannot load new structures data in production.`) which is a 500 error

are we expecting something else here?

Is this something like the handler is already populated when I started the dev server.

Reply:
Hmm. @brianrodri / @vojtechjelinek is it not possible to conditionally register this route based on dev mode? Vojta, you mentioned an alternative way to gate this, but I'm wondering why we can't just conditionally add a dev-only list of routes based on whether dev-mode is enabled.

I think the approach @DubeySandeep mentioned makes sense--if it works. I'm a bit confused why @FareesHussain is seeing a 500 error though--any ideas?

/cc @seanlip as well in case you have thoughts or want to weigh in.

Reply:
Hmm I actually lean more towards raising an unauthorized exception in this situation, something like:

```python
if not constants.DEV_MODE:
    raise self.UnauthorizedException()
```

Seems cleaner/simpler than conditionally changing our handlers.

Reply:
Re the 500 error, there's not enough info given. I'd need the stack trace of the 500 error in order to weigh in.

I'm ok with @brianrodri's suggestion too but shouldn't we just raise a 404? Unauthorized (401) is for when the user doesn't have the right permissions, which seems incorrect in this case.

Reply:
I think sandeep's suggestion will give a 404 in the production server, while running the tests it looks like the route is registered before the running the test case and hence it doesn't give a 404 here.

Is it possible to tests a route in the test server (https://oppiatestserver.appspot.com/) using the sandeep's suggestion.

WDYT?

Reply:
> Is it possible to tests a route in the test server

Depends on what you want to test! IF the rout is not register then it will throw 404*